src/Messages.jl

```diff
@@ -609,7 +609,23 @@ function Base.show(io::IO, m::Message)
     end
     println(io, typeof(m), ":")
     println(io, "\"\"\"")
-    writeheaders(io, m)
+
+    # Mask sensitive header values
+    # The following headers values contain sensitive information that
+    # we don't want to show
+    # - Authorization
+    # - Cookie
+    # - Set-Cookie (in response)
+    # We will show "**********" instead
+    ioh = IOBuffer()
+    writeheaders(ioh, m)
+    header_str = String(take!(ioh))
+    mask_headers = ["Authorization", "Cookie", "Set-Cookie"]
+    for mh in mask_headers
+        header_str = replace(header_str, Regex("($mh: ).*\$", "m") => s"\1******")
```

Comment:
```suggestion
        header_str = replace(header_str, Regex("($mh: ).*\$", "mi") => s"\1******")
```
not sure if we're likely to get differently cased headers, but making the regex case insensitive seems like a good idea regardless.

Reply:
Should also match the start of a line, and only match `\r\n` line breaks, I think:
```suggestion
        header_str = replace(header_str, Regex("(*CRLF)^($mh: ).*\$", "mi") => s"\1******")
```

Perhaps also show `<REDACTED>` or something instead of `*****` to make it more obvious that it is not the actual header value that is shown? On the other hand, `****` is widely used to mask things like this.